### ISSUE TITLE
Syntax error with CircuitPy 7.3.3 and the PyRuler default program

### ISSUE THREAD

**Issue by mschmitt3459** [adafruit/circuitpython-default-files#11]
Apologies if I'm using this Issues functionality incorrectly.  I have a new PyRuler that I updated to firmware v7.3.3.  Now, the default code.py program that ships with the ruler throws a syntax error in Mu.  I made my way over to this code repository.  I suspect the v4 and v6 folders refer to CircuitPython versions.  If so, I suspect my issue is that something changed in v7 that necessitates a syntax update in the default program.  

Mu indicates the issue is in line 31:  touches = [DigitalInOut(board.CAP0)]

  File "code.py", line 31, in <module>
AttributeError: 'module' object has no attribute 'CAP0'

So I gather that the board library no longer refers to Capacitive Touch 0 as CAP0.   At this point, I'm at the end of my abilities.  I'm not sure where to go to see the documentation for the board library so I can attempt to update the code. 

Any help you could provide would be greatly appreciated!

Thanks.

Micah

**Comment by tannewt** [comment]
I still see CAP0 listed in the source code. In the REPL can you do `dir(board)` to see what pins it has? Did you accidentally install a different board's firmware?

**Comment by mschmitt3459** [comment]
Scott,

Thanks for responding!  I don' t think that I installed the wrong
firmware.  I used the 7.3.3 version found here
<https://circuitpython.org/board/trinket_m0/>.  In the REPL, a dir(board)
returns a syntax error:

Auto-reload is on. Simply save files over USB to run them or enter REPL to
disable.


Press any key to enter the REPL. Use CTRL-D to reload.


Adafruit CircuitPython 7.3.3 on 2022-08-29; Adafruit Trinket M0 with
samd21e18

>>> dir(board)

Traceback (most recent call last):

File "<stdin>", line 1, in <module>

NameError: name 'board' is not defined

**Comment by kattni** [comment]
Hello, @mschmitt3459.

Trinket M0 is not the appropriate CircuitPython build for the PyRuler. It has its own build found here: https://circuitpython.org/board/pyruler/ 

Please try downloading the proper version and see if that helps.

Further, you need to `import board` before running `dir(board)` in the REPL.

Hope this helps!

**Comment by mschmitt3459** [comment]
Thanks!  It's working now.

For what it's worth, I followed this path which lead me to the wrong
firmware:

From the Adafruit PyRuler product page, I read the following paragraph and
clicked the link:

[image: image.png]

Then, I saw the colored boxes at the beginning of the Circuit Python page
indicating I should update.  So I followed the instructions on that page.
That took me to the non-PyRuler Trinket M0 download.

[image: image.png]
